backend/src/data/repositories/course/course.repository.ts

```diff
@@ -17,15 +17,20 @@ class Course {
 
   public getAll(filteringOpts: {
     categoryId: number | null;
+    title: string;
   }): Promise<CourseGetResponseDto[]> {
-    const { categoryId } = filteringOpts ?? {};
+    const { categoryId, title } = filteringOpts ?? {};
 
     return this.#CourseModel
       .query()
       .where((builder) => {
         if (categoryId) {
           builder.where({ courseCategoryId: categoryId });
         }
+
+        if (title) {
+          builder.where('title', 'ilike', `%${title}%`);
+        }
       })
```

Comment:
Split up different where conditions to separate statements
```suggestion
      .where((builder) => {
        if (categoryId) {
          builder.where({ courseCategoryId: categoryId });
        }
      })
      .andWhere((builder) => {
        if (categoryId) {
          builder.where({ courseCategoryId: categoryId });
        }
      })
```